File: src/leetcode/m11/ContainerWithMostWater.java
/*
 * Copyright (c) 2025 THGEEK.COM. All rights reserved.
 */

package leetcode.m11;

/**
 * LeetCode 11
 *
 * @author Tao Hong
 * @version 1.0
 * @since 2025/02/03 13:05
 */
public class ContainerWithMostWater {
    public int maxArea(int[] height) {
        int max = 0;
        for (int i = 0, j = height.length - 1; i < j; ) {
            int minHeight = height[i] < height[j] ? height[i++] : height[j--];
            max = Math.max(max, minHeight * (j - i + 1));
        }
        return max;
    }
}

File: src/leetcode/m11/ContainerWithMostWater.go
package m11

// LeetCode 11
func maxArea(height []int) int {
	res := 0
	for i, j := 0, len(height)-1; i < j; {
		var minHeight int
		if height[i] < height[j] {
			minHeight = height[i]
			i++
		} else {
			minHeight = height[j]
			j--
		}

		res = maxInt(minHeight*(j-i+1), res)
	}
	return res
}

func maxInt(a, b int) int {
	if a < b {
		return b
	}
	return a
}
